src/common/tracer.cc

```diff
@@ -21,10 +23,11 @@ Tracer::Tracer(opentelemetry::nostd::string_view service_name) {
 
 void Tracer::init(opentelemetry::nostd::string_view service_name) {
   if (!tracer) {
-    const opentelemetry::exporter::jaeger::JaegerExporterOptions opts;
-    auto jaeger_exporter  = std::unique_ptr<opentelemetry::sdk::trace::SpanExporter>(new opentelemetry::exporter::jaeger::JaegerExporter(opts));
-    auto processor = std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>(new opentelemetry::sdk::trace::SimpleSpanProcessor(std::move(jaeger_exporter)));
+    const opentelemetry::exporter::jaeger::JaegerExporterOptions exporter_options;
+    const opentelemetry::sdk::trace::BatchSpanProcessorOptions processor_options;
     const auto jaeger_resource = opentelemetry::sdk::resource::Resource::Create(std::move(opentelemetry::sdk::resource::ResourceAttributes{{"service.name", service_name}}));
+    auto jaeger_exporter = std::unique_ptr<opentelemetry::sdk::trace::SpanExporter>(new opentelemetry::exporter::jaeger::JaegerExporter(exporter_options));
+    auto processor = std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>(new opentelemetry::sdk::trace::BatchSpanProcessor(std::move(jaeger_exporter), processor_options));
```

Comment:
@zenomri what improvement does batchspanprocesser bring?

Reply:
performance. it stores the spans until the timeout is reached (5s) or the buffer fills up. the simple processor just sends a span in their d'tor

Reply:
@zenomri what was the difference in the performance numbers?

Reply:
@yuvalif 
with simple span processor: http://folio08.front.sepia.ceph.com/1502/hsbench_output_simple.txt
50% drop in PUT and 55% drop in GET operations, compared to the batched processor.

Reply:
batched results: http://folio08.front.sepia.ceph.com/1502/hsbench_output_enabled.txt (copied from the decsription)

Reply:
great, thanks @yuvalif @zenomri !